File: 13-02-2026/problem_10_merge_k_arrays/correct/merge_k_arrays.py
# PROBLEM 3: MERGE K SORTED ARRAYS - CORRECT PYTHON
# ===================================================
import sys

K = int(sys.stdin.readline())

all_elements = []

for _ in range(K):
    line = list(map(int, sys.stdin.readline().split()))
    n = line[0]
    elements = line[1:n + 1]
    all_elements.extend(elements)

all_elements.sort()

print(*all_elements)

File: 13-02-2026/problem_10_merge_k_arrays/correct/merge_k_arrays.java
// PROBLEM 3: MERGE K SORTED ARRAYS - CORRECT JAVA
// ===================================================
import java.util.*;

class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        if (!sc.hasNextInt()) return;
        int K = sc.nextInt();
        
        ArrayList<Integer> all = new ArrayList<>();
        
        for (int i = 0; i < K; i++) {
            int n = sc.nextInt();
            for (int j = 0; j < n; j++) {
                all.add(sc.nextInt());
            }
        }
        
        Collections.sort(all);
        
        for (int i = 0; i < all.size(); i++) {
            if (i > 0) System.out.print(" ");
            System.out.print(all.get(i));
        }
        System.out.println();
        
        sc.close();
    }
}